client/blocks/image-editor/index.jsx

```diff
@@ -75,11 +92,38 @@ const ImageEditor = React.createClass( {
 			this.props.resetAllImageEditorState();
 
 			this.updateFileInfo( newProps.media );
+
+			this.setDefaultAspectRatio();
 		}
 	},
 
 	componentDidMount() {
 		this.updateFileInfo( this.props.media );
+
+		this.setDefaultAspectRatio();
+	},
+
+	setDefaultAspectRatio() {
+		const {
+			defaultAspectRatio: propsDefaultAspectRatio,
+			allowedAspectRatios
+		} = this.props;
+
+		let defaultAspectRatio = propsDefaultAspectRatio;
+
+		// If defaultAspectRatio was not specified or is not included in allowedAspectRatio.
+		if ( indexOf( allowedAspectRatios, propsDefaultAspectRatio ) === -1 ) {
+			defaultAspectRatio = get( allowedAspectRatios, '0', -1 );
+		}
+
+		// If default aspect ratio is not valid.
+		if ( ! AspectRatios[ defaultAspectRatio ] ) {
```

Comment:
I thought we checked for defaultAspectRatio type in PropTypes. Can you think of a case where we would hit this?